e2e/setup/packager.js

```diff
@@ -0,0 +1,68 @@
+/*
+MIT License
+
+Copyright (c) 2021-present, Elastic NV
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+const { spawn } = require('child_process');
+const { env } = require("../services");
+
+const packageFiles = async () => {
+  return new Promise((resolve, reject) => {
+    const ls = spawn('npm', ["run", "react:start"], {
+      env: {
+        PORT: env.TEST_PORT,
+        PATH: process.env.PATH,
+        BROWSER: "none",
+      },
+      shell: true,
+      detached: true
+    });
+
+
+    ls.stdout.on('data', async (data) => {
+      if (data.indexOf("Something is already running on port") !== -1) {
```

Comment:
I like the idea of using a dynamic port but I'm not sure there are significant advantages to it. Please let me know what you think about the following points:

1. Using a dynamic port could easily mean that users would end up with lots of packager processes running. That could happen because packager process would still be running if the `node` process with tests crashed without the opportunity to cleanup (or got killed).
2. In the future, I'd like to allow users to start the packager separately using this environment variable and then on every run we'd simply spin up the runner (the client). That's similar to what happens in Kibana where you have the functional test runner and the functional test server. This makes debugging easier and quicker. If we were to use a dynamic port you'd have to pass that to the runner command every time, which is fine, but not ideal I'd say.
3. For us to use a dynamic port we will have to persist the state to the underlying system, otherwise Jest can't access the PID of the process spawned during the `setupFiles` phase within test files and thus can't make the electron instances point to the packager. Storing that state in the system would also solve (1) and (2), but IMO it's too much extra complexity.
    An alternative would be to move the setup and teardown of the packager into Jest hooks, but then that's even less clean IMO as we'd be using test hooks to setup global processes (this is just aesthetic, but thought I'd mention).
4. Overall, the dynamic port adds unnecessary complexity IMO, as with a fixed port it's easier for users to simply know what will happen when they start their tests.

Considering the reasons above, and that users can simply change the port by setting the `TEST_PORT` variable if they need to, I'm not sure we should have a dynamic port.

What are your thoughts on this?